### DISCUSSION TITLE
Appending kilosort and phy results in pynwb

### DISCUSSION THREAD

**Discussion by RuthKassahun** [NeurodataWithoutBorders/helpdesk#60] [Q&A] [answered]
Hello,
I was attempting to store the output I got from kilosort and phy in an existing nwb file. I followed this [documentation ](https://pynwb.readthedocs.io/en/stable/pynwb.ecephys.html#pynwb.ecephys.Clustering). I got value error as the data aren't in the same shape. How can I create a clustering interface? Thank you!
```
clustering = Clustering(
                       description='Kilosort outputs', # description
                       num=spike_clusters, # cluster ID
                       peak_over_rms= amplitudes, #(not quite) peak over RMS
                       times=spike_times / sampling_rate, # times in s
                       name='clustering') # name
```
`ValueError: Clustering.__init__: incorrect shape for 'num' (got '(5664215, 1)', expected '(None,)'), incorrect shape for 'peak_over_rms' (got '(5664215, 1)', expected '(None,)'), incorrect shape for 'times' (got '(5664215, 1)', expected '(None,)'`

**Comment by CodyCBakerPhD** [comment]
As mentioned in the description of that documentation, the data type as been deprecated for quite some time - it is recommended to use a `Units` table instead

Tutorial for spike timing data across units: https://pynwb.readthedocs.io/en/stable/tutorials/domain/ecephys.html#sphx-glr-tutorials-domain-ecephys-py

**Comment by rly** [comment, marked as answer]
If you have output from kilosort and phy, you can also use Neuroconv to append the converted output of kilosort/phy to an existing NWB file:

https://neuroconv.readthedocs.io/en/main/conversion_examples_gallery/sorting/kilosort.html
https://neuroconv.readthedocs.io/en/main/conversion_examples_gallery/sorting/phy.html
